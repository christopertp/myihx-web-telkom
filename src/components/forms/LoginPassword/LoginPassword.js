import React from 'react';
import PropTypes from 'prop-types';
import { useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import { Field } from 'redux-form';
import Button from '../../elements/Button';
import Spinner from '../../elements/Spinner';
import PasswordField from '../../fields/Password';
import styles from './styles.scoped.css';

export default function LoginPassword(props) {
  const { handleSubmit, invalid } = props;
  const { isLoading } = useSelector(s => s.login);
  const inputProps = {
    autoFocus: true,
    label: 'KATA SANDI',
    placeholder: 'Masukkan kata sandi Anda',
  };

  return (
    <form className={styles.root} onSubmit={handleSubmit}>
      <Field component={PasswordField} inputProps={inputProps} name="password" />
      <Link to="/forgot-password">Lupa kata sandi</Link>
      <Button disabled={invalid || isLoading} type="submit">
        {isLoading ? <Spinner /> : 'Lanjut'}
      </Button>
    </form>
  );
}

LoginPassword.defaultProps = {
  handleSubmit: () => {},
  invalid: false,
};

LoginPassword.propTypes = {
  handleSubmit: PropTypes.func,
  invalid: PropTypes.bool,
};
