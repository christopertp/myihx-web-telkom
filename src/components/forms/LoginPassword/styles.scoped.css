.root > a {
  color: var(--primary-color);
  display: block;
  font-weight: 800;
  font-size: 0.9rem;
  margin-top: 1rem;
  text-align: right;
  text-transform: uppercase;
}

.root > button:last-of-type {
  margin-top: 2rem;
  width: 100%;
}
