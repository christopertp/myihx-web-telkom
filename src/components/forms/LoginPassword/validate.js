export default function validate(values) {
  const { password } = values;

  return {
    password: !password ? 'Harus diisi' : '',
  };
}
