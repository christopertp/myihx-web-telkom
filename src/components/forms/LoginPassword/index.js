import { reduxForm } from 'redux-form';
import LoginPassword from './LoginPassword';
import validate from './validate';

export default reduxForm({
  initialValues: {
    password: '',
  },
  form: 'loginPassword',
  validate,
})(LoginPassword);
