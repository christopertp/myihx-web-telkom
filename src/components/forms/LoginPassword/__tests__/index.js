import React from 'react';
import { useSelector } from 'react-redux';
import ShallowRenderer from 'react-test-renderer/shallow';
import Spinner from '../../../elements/Spinner';
import LoginPassword from '../LoginPassword';

useSelector.mockImplementation(fn => {
  fn(({ login: '' }));
  return { isLoading: false, message: '' };
});

describe('src/components/forms/LoginPassword', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<LoginPassword />);
    expect(tree).toMatchSnapshot();

    useSelector.mockImplementationOnce(() => ({ isLoading: true, message: 'error' }));
    LoginPassword.defaultProps.handleSubmit();
    const result = LoginPassword({
      ...LoginPassword.defaultProps,
    });
    const button = result.props.children[2];
    expect(button.props.children.type).toBe(Spinner);
  });
});
