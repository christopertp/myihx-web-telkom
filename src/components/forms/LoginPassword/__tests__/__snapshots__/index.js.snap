// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/forms/LoginPassword render 1`] = `
<form
  onSubmit={[Function]}
>
  <Field
    component={[Function]}
    inputProps={
      Object {
        "autoFocus": true,
        "label": "KATA SANDI",
        "placeholder": "Masukkan kata sandi Anda",
      }
    }
    name="password"
  />
  <Link
    to="/forgot-password"
  >
    Lupa kata sandi
  </Link>
  <Button
    className=""
    disabled={false}
    onClick={null}
    to=""
    type="submit"
    variant="red"
  >
    Lanjut
  </Button>
</form>
`;
